pkg/discovery/mesos_leader.go

```diff
@@ -0,0 +1,102 @@
+package discovery
+
+import (
+	"github.com/turbonomic/mesosturbo/pkg/conf"
+	master "github.com/turbonomic/mesosturbo/pkg/masterapi"
+	"fmt"
+	"strconv"
+	"strings"
+	"github.com/golang/glog"
+	"github.com/turbonomic/mesosturbo/pkg/data"
+)
+//
+//func main() {
+//	ipportlist := "10.10.174.92:5050,10.10.174.91:5050,10.10.174.100, 10.10.174.101:5050 "
+//	urlList := ParseMasterIPPorts(ipportlist)
+//	masteruser := "enlin.xu"
+//	masterpwd := "sysdreamworks"
+//	NewMesosCluster(conf.Apache, urlList, masteruser, masterpwd)
+//}
+//
+type MesosMasterUrl struct {
+	IP string
+	Port string
+}
+
+type MesosLeader struct {
+	leaderConf *conf.MesosTargetConf
+	MasterState *data.MesosAPIResponse
+}
+
+func NewMesosLeader(mesosMasterType conf.MesosMasterType, urlList []*MesosMasterUrl, masteruser, masterpwd string) (*MesosLeader, error) {
```

Comment:
Does function NewMesosLeader() have two purpose:
  (1) get the leader;
  (2) get the status of the cluster, including status of the tasks;

If its purpose is only to get the leader, you can send a request to any of the master nodes with a url like  "http://IP:5050/redirect".  The header of the response will contain  an item "Location", which is the url of the leader.